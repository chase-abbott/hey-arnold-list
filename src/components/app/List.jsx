import React from 'react';
import ListItem from './ListItem';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Ul = styled.ul`
list-style-type: none;
display: grid;
grid-template-columns: 1fr 1fr 1fr 1fr;
`; 

const Li = styled.li`
display: flex;
justify-content: center;
margin: 25px;
`;

export default function List({ characters }){
  return (
    <Ul data-testid="list">
      {characters.map(char => {
        return (
          <Li key={char._id}>
            <ListItem image={char.image} id={char._id}/>
          </Li>
        ); 
      })}
    </Ul>
  ); 
}

List.propTypes = {
  characters: PropTypes.arrayOf(
    PropTypes.shape({
      _id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired
    })
  )
};
