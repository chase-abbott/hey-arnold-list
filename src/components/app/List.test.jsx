/* eslint-disable max-len */
import React from 'react';
import { render } from '@testing-library/react';
import List from './List';

describe('List item component', () => {
  it('renders a list item to the page', () => {
    const characters = [{
      '_id': '5da237699734fdcb7bef8fb8',
      'name': 'Shelley',
      'image': 'https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256'
    }];
    
    const { asFragment } = render(<List characters={characters}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
