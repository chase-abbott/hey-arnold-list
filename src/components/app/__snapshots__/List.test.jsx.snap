// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List item component renders a list item to the page 1`] = `
<DocumentFragment>
  <ul
    class="sc-gtsrHT tQpUX"
  >
    <li
      class="sc-dlnjwi hftDXS"
    >
      <a
        href="/detail/5da237699734fdcb7bef8fb8"
      >
        <img
          alt="image"
          class="sc-bdnxRM gagOgl"
          src="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
        />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
